### PR TITLE
[Do Not Merge] Airtable

### PR DESCRIPTION
**Pull request by thchia**
Testing e2e.

### CODE REVIEW

**Comment by thchia** [comment]
@wgao19 If you want to try, there is a small RSVP widget on the next talk pane on the index page, and also on the meetup page.

- Should we allow people to "hide" their attendance?
- I haven't done anything to select whether you are `Speaker` or `Attendee`, it's just hardcoded to `Attendee` right now
- I still need to tidy up a lot of UX stuff
- Do we need to collect any more fields?
- I will tidy up the git history before we merge this

Edit: Also, this will potentially make it harder to run locally, as it needs an Airtable API key now. I wonder if we can opt out of running the airtable queries somehow.

**Comment by thchia** [comment]
Thoughts from me on the code. For no good reason I have:

- not used a form library (though honestly I think it’s overkill for this)
- not used netlify-lambda/netlify-dev to build the function code. Instead there is a nested package in src/functions with the necessary deps for the function code.

File: src/functions/airtable.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import Airtable from 'airtable';
+
+export const handler = async (event, _, callback) => {
+  try {
+    let attendees = [];
```
Comment (thchia) [inline src/functions/airtable.js]: This is ghost code

File: src/components/RSVP/index.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import axios from 'axios';
+import s from './s.module.scss';
+
+const initialState = {
+  name: '',
+  username: '',
+  validationError: '',
```
Comment (thchia) [inline src/components/RSVP/index.js]: Unnecessary

File: src/functions/airtable.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            'Created Date': new Date().toISOString(),
+          },
+        },
+      ])
+    callback(null, {
+      status: 200,
+      body: JSON.stringify(attendees),
+    });
+  } catch (e) {
+    callback(null, {
```
Comment (thchia) [inline src/functions/airtable.js]: Probably should have something in the first arg.

File: src/templates/Meetup/index.jsx [hunk 4]
```diff
@@ -215,6 +218,13 @@ export default ({
             </a>
           </SubmitTalkButton>
         </section>
+        <section>
+          <h2><span role="img" aria-label="busts in silhouette">👥</span> Attendees <span role="img" aria-label="busts in silhouetee">👥</span></h2>
```
Comment (wgao19) [inline src/templates/Meetup/index.jsx]: lol til the name of this emoji 👥

File: src/pages/index.jsx [hunk 5]
```diff
@@ -72,15 +73,11 @@ export default ({
...
-                <p>
-                  <b>
-                    {eventLink ? (
-                      <a href={eventLink}>RSVP</a>
-                    ) : (
-                      'RSVP open soon'
-                    )}
-                  </b>
-                </p>
+                {eventLink ? (
```
Comment (wgao19) [inline src/pages/index.jsx]: do we still need to check `eventLink`

File: src/functions/airtable.js (hunk 1 above)
Comment (wgao19) [reply src/functions/airtable.js]: 👻

**Comment by wgao19** [comment]
i just checked with michael, he asks for only names, and [because of pdpa](https://singaporelegaladvice.com/law-articles/essential-pdpa-compliance-guide-singapore-businesses/), we cant store personal information on our site, (github handle is ok, not sure about email, but we dont have to ask this until a venue specifically asks us for)

so i think the fields we have now is good

File: src/pages/index.jsx (hunk 5 above)
Comment (thchia) [reply src/pages/index.jsx]: No need

**Pull request closed without merging**